packages/docgen/src/engine.js

```diff
@@ -2,15 +2,14 @@
  * External dependencies
  */
 const babel = require( '@babel/core' );
-const { flatten } = require( 'lodash' );
 
 /**
  * Internal dependencies
  */
 const getIntermediateRepresentation = require( './get-intermediate-representation' );
 
-const getAST = ( source ) => {
-	return babel.parse( source || '' ).program;
+const getAST = ( source, filename ) => {
```

Comment:
Reading the issue description I gather we use this to signal the `babel-preset-typescript` babel is processing a tsx file, but I haven't found info about that [here](https://babeljs.io/docs/en/babel-parser) or [here](https://babeljs.io/docs/en/babel-preset-typescript). Just curiosity, but it'd be nice to have some pointers to this.

Reply:
Here we're not using the `@babel/parser` module directly, but indirectly through the `@babel/core` module which exports a `parse` function. `@babel/core` loads the Babel config will all the plugins and presets, and then exposes a configured parser.

This `parse` method is documented [here](https://babeljs.io/docs/en/babel-core#parsesync) and the `filename` option is documented [here](https://babeljs.io/docs/en/options#filename).

If the `filename` option is not present, the parsing will fail with error:
> Error: [BABEL] unknown: Preset /* your preset */ requires a filename to be set when babel is called directly,
> ```
> babel.transform(code, { filename: 'file.ts', presets: [/* your preset */] });
> ```
> See https://babeljs.io/docs/en/options#filename for more information.
>    at validateIfOptionNeedsFilename (/Users/jsnajdr/src/gutenberg/node_modules/@babel/core/lib/config/full.js:286:11)